internal/terraform/node_resource_abstract_instance.go

```diff
@@ -584,8 +584,9 @@ func (n *NodeAbstractResourceInstance) writeChange(ctx EvalContext, change *plan
 }
 
 // refresh does a refresh for a resource
```

Comment:
nit: can you update the docs here just clarifying that a nil deferred object means no deferral and anything else means we are deferring.